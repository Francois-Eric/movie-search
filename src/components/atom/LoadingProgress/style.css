.LoadingProgress {
  display: table;
  margin-bottom: 2rem;
}
