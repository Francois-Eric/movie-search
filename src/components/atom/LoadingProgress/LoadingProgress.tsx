import { CircularProgress, LinearProgress, Typography } from "@mui/material";
import "./style.css";

interface Props {
  count: number;
  total: number;
}

export const LoadingProgress = ({ count, total }: Props) => {
  return (
    <div className={"LoadingProgress"}>
      {count > 0 ? (
        <>
          <Typography variant="subtitle2" display="block">
            {`${count} résultat${count > 0 && "s"} chargé${
              count > 0 && "s"
            } sur ${total}`}
          </Typography>
          <LinearProgress
            variant="determinate"
            value={(count / total) * 100 || 0}
          />
        </>
      ) : (
        <Typography variant="subtitle2" display="block">
          Aucun résultat
        </Typography>
      )}
    </div>
  );
};
