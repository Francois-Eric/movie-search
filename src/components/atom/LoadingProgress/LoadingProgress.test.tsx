import { render } from "@testing-library/react";
import { LoadingProgress } from "./LoadingProgress";

describe("<LodingProgress/>", () => {
  test("should render correctly", () => {
    const { container } = render(<LoadingProgress count={50} total={100} />);
    expect(container).toMatchSnapshot();
  });
});
