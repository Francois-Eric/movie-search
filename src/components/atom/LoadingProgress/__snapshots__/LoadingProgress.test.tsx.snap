// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LodingProgress/> should render correctly 1`] = `
<div>
  <div
    class="LoadingProgress"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 css-1sr7afs-MuiTypography-root"
    >
      50 résultats chargés sur 100
    </h6>
    <span
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="50"
      class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-eglki6-MuiLinearProgress-root"
      role="progressbar"
    >
      <span
        class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-5xe99f-MuiLinearProgress-bar1"
        style="transform: translateX(-50%);"
      />
    </span>
  </div>
</div>
`;
